### DISCUSSION TITLE
How to add FixedPixelCircle (mapsforge) in VTM?

### DISCUSSION THREAD

**Discussion by mahdian-organization** [mapsforge/vtm#861] [Q&A] [answered]
Hi,
I want to create a point layer like mapsforge's FixedPixelCircle. This is my code:
`VectorLayer vectorLayer = new VectorLayer(map);
vectorLayer.add(new PolygonDrawable(geometryBuilder.get(), defaultStyle()));

            vectorLayer.add(new PointDrawable(tapLatLong, Style.builder()
                .buffer(0.5)
                .fillColor(Color.RED)
                .fillAlpha(0.2f).fixed(true)
                .build()));

            map.layers().add(vectorLayer);
            vectorLayer.update();
`

But circles is very large.

**Comment by devemux86** [comment, marked as answer]
Use the [Style.scaleZoomLevel](https://github.com/mapsforge/vtm/blob/master/vtm-jts/src/org/oscim/layers/vector/geometries/Style.java#L191) option to specify from which zoom level the point should stop decreasing in size and "stick to the map".

**Reply by mahdian-organization** [reply]
Thank you very much
But I want to circle size to be equal in all zoom levels. Like "FixedPixelCircle"
When I use `style.scaleZoomLevel(8)` circle size is very small in 5 zoom level and size is very big in more than 10.

**Reply by devemux86** [reply]
> When I use style.scaleZoomLevel(8) circle size is very small in 5 zoom level and size is very big in more than 10.

Then use a very large `scaleZoomLevel` like 20.

**Reply by mahdian-organization** [reply]
Thanks @devemux86,
Point is very small in `scaleZoomLevel=20` when zoom level is 4 or 5.

**Reply by devemux86** [reply]
> Point is very small in scaleZoomLevel=20 when zoom level is 4 or 5.

Please study the [Style](https://github.com/mapsforge/vtm/blob/master/vtm-jts/src/org/oscim/layers/vector/geometries/Style.java) API options.
If you want a larger circle then increase the `buffer`.

Or even better use [markers](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/test/MarkerOverlayActivity.java) with a circle symbol.

Or as mentioned above you can use your own [circle renderer](https://github.com/mapsforge/vtm/blob/master/vtm-android-example/src/org/oscim/android/cluster/ClusterUtils.java).

**Comment by demiantres** [comment]
You can check the (clustered) marker example in the project. The markers are circles which do not change their size.
